wire-compiler/src/main/java/com/squareup/wire/java/TypeWriter.java

```diff
@@ -506,6 +550,285 @@ private MethodSpec messageHashCode(MessageType type) {
     return result.build();
   }
 
+  // Example:
+  //
+  // @Override
+  // public String toString() {
+  // }
+  private MethodSpec messageToString(MessageType type, ClassName typeName) {
+    MethodSpec.Builder result = MethodSpec.methodBuilder("toString")
+        .addAnnotation(Override.class)
+        .addModifiers(PUBLIC)
+        .returns(String.class);
+
+    result.addStatement("$1T sb = new $1T()", StringBuilder.class);
+
+    for (Field field : type.fieldsAndOneOfFields()) {
+      String methodName = field.options().optionMatches(".*\\.redacted", "true")
+          ? "toStringRedacted" : "toStringValue";
+      result.addStatement("$L(sb, $S, $L)", methodName, field.name(),
+          sanitize(field.name()));
+    }
+
+    if (!type.extensions().isEmpty()) {
+      result.addStatement("toStringValue(sb, $S, extensionsToString())", "{extensions");
+    }
+
+    result.addStatement("sb.insert(0, $S)", typeName.simpleName() + "{");
```

Comment:
`toStringValue` checks against `.length() > 0` to determine if it prepends a comma. If this was appended first it would need to take an argument with the length of this initial string to compare against (which was the first implementation, actually, before switching to this).